pkg/lib/algorithm/iblt/sync.go

```diff
@@ -1,14 +1,41 @@
 package iblt
 
 import (
+	"fmt"
+	"math"
+
+	"github.com/SheldonZhong/go-IBLT"
+	"github.com/sirupsen/logrus"
+
 	"github.com/String-Reconciliation-Ditributed-System/RCDS_GO/pkg/lib/genSync"
 	"github.com/String-Reconciliation-Ditributed-System/RCDS_GO/pkg/set"
+	"github.com/String-Reconciliation-Ditributed-System/RCDS_GO/pkg/util"
 )
 
-type ibltSync genSync.SetSyncInfo
+type ibltSync struct {
+	*iblt.Table
+	*set.Set
+	FreezeLocal   bool
+	SentBytes     int
+	ReceivedBytes int
+}
+
+func NewIBLTSetSync(diffNum, dataLen int) (genSync.GenSync, error) {
+	if diffNum <= 0 {
+		return nil, fmt.Errorf("number of difference should be positive")
+	}
+	// TODO: table size should be 1.5 times the sym set diff, IBLT algo impl should improve.
+	tableSize := diffNum * 4
+	if tableSize < 4 {
+		tableSize = 4
+	}
```

Comment:
why is this necessary?

Reply:
Good question. The IBLT table size is (set to) linear to the number of symmetrical set differences between the sets. Theoretically the factor should be 1.5 but we have 4 here, I think I found the reason and will update to see if it can be cut down to a smaller number than 4. Anyway, this is to say that the minimum table size should be 4. It is a “magical” number but this number should be at least 2 for resolving 1 diff given that the minimum number of hash fxns should be 2. The hash fxn is to assign where the value sits in the table. The minimum has to be 2 to enable peeling process.